gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/MediaUtil.java

```diff
@@ -369,288 +339,362 @@ public static void setMetaData(final FileHandler fileHandler, final Photo photo,
+            nameAuthor = defaultStringIfNotDefined(getWatermarkValue(watermarkHD), nameAuthor);
+            if (!nameAuthor.isEmpty()) {
+              final HandledFile watermarkFile = getHandledFile().getParentHandledFile()
+                  .getHandledFile(photo.getId() + "_watermark.jpg");
+              AbstractImageToolOption option = WatermarkTextOption.text(nameAuthor);
+              getImageTool().convert(getHandledFile().getFile(), watermarkFile.getFile(), option);
             }
           }
-        }
-        if (StringUtil.isDefined(watermarkOther)) {
-          final String value = getWatermarkValue(watermarkOther, iptcMetadata);
-          if (value != null) {
-            nameAuthor = value;
-          }
-          if (!nameAuthor.isEmpty()) {
-            nameForWatermark = nameAuthor;
+          if (isDefined(watermarkOther)) {
+            nameAuthor = defaultStringIfNotDefined(getWatermarkValue(watermarkOther), nameAuthor);
+            if (!nameAuthor.isEmpty()) {
+              nameForWatermark = nameAuthor;
+            }
           }
+        } catch (MediaMetadataException e) {
+          SilverLogger.getLogger(MediaUtil.class).error(
+              "Bad image file format " + getHandledFile().getFile().getPath() + ": " +
+                  e.getMessage());
+        } catch (UnsupportedEncodingException e) {
+          SilverLogger.getLogger(MediaUtil.class).error(
+              "Bad metadata encoding in image " + getHandledFile().getFile().getPath() + ": " +
+                  e.getMessage());
         }
-      } catch (MediaMetadataException e) {
-        SilverLogger.getLogger(MediaUtil.class)
-            .error("Bad image file format " + image.getFile().getPath() + ": " + e.getMessage());
-      } catch (UnsupportedEncodingException e) {
-        SilverLogger.getLogger(MediaUtil.class).error(
-            "Bad metadata encoding in image " + image.getFile().getPath() + ": " + e.getMessage());
       }
+      return nameForWatermark;
+    }
+
+    /**
+     * Gets lazily the IPTC data from a photo.
+     * @return
+     * @throws MediaMetadataException
+     * @throws IOException
+     */
+    private List<MetaData> getIptcMetaData() throws MediaMetadataException, IOException {
+      if (iptcMetadata == null) {
+        final MediaMetadataExtractor extractor =
+            new DrewMediaMetadataExtractor(getMedia().getInstanceId());
+        iptcMetadata = extractor.extractImageIptcMetaData(getHandledFile().getFile());
+      }
+      return iptcMetadata;
     }
-    return nameForWatermark;
-  }
 
-  private static String getWatermarkValue(final String property,
-      final List<MetaData> iptcMetadata) {
-    String value = null;
-    for (final MetaData metadata : iptcMetadata) {
-      if (property.equalsIgnoreCase(metadata.getProperty())) {
-        value = metadata.getValue();
+    private String getWatermarkValue(final String property) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)